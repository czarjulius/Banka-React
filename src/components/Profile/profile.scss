a:hover {
    font-size: 12px;
    transition: 0.5s;
    }
    .profile-container{
      margin: auto;
      width: 60%;
      height: auto;
      text-align: center;
      color: gray
    }
    .personal-detail-container{
      padding: 30px;
      width: 98%;
      height: auto;
      display: flex;
      margin-bottom: 20px;
      border-radius: 8px;
      box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  
    }
  
    .fa{
      padding: 10px;
      color: rgb(48, 127, 138);
      }
      
    .profile-image{
      width: 35%;
      height: auto;
      margin-right: 35px;
    }
    .profile-detail{
      padding: 10px;
      width: 60%;
      height: auto;
    }
    .account-detail-container{
      padding: 30px;
      width: 98%;
      height: auto;
      border-radius: 8px;
      box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  
    }
  .details-container {
    display: flex;
    font-family: 'Times New Roman', Times, serif;
    flex-wrap: wrap;
  }
  .profile-pic {
    padding: 30px;
    flex: 1;
  }
  .personal-detail {
    flex: 2;
    padding: 30px;
  }
  .account-detail {
    flex: 2;
    padding: 30px;
    margin-bottom: 20px;
  }
  .profile-pic img{
  width: 200px;
  height: 200px;
  
  } 
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
  }
  
  th, td {
    text-align: right;
    /* padding: 10px; */
    font-size: 15px;
  }
  
  th {
    color: rgb(3, 26, 54);
  }
  
  th:first-child, td:first-child {
    text-align: left;
  }
  
  caption{
    font-family: 'Times New Roman', Times, serif;
    font-size: 20px;
    text-align: center;
    padding: 10px;
    text-decoration: underline;
  }
  .fa-check {
    color: rgb(116, 187, 9);
  }
  .fa-map-marker{
    color: rgb(170, 7, 7);
  }
  .fa-calendar-o{
    color: rgb(170, 7, 7);
  }
  .fa-envelope{
    color: rgb(157, 200, 240);
  }
  .fa-user{
    color: rgb(157, 200, 240);
  }
  
  @media screen and (max-width: 1024px) {
    .personal-detail-container{
      flex-wrap: wrap;
    }
    .profile-detail{
      width: 90%;
    }
    .profile-container{
      width: 90%;
    }
  }
  
  