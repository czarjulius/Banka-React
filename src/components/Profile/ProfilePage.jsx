import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import Dashboard from '../Dashboard/DashboardPage';
import './profile.scss';
import { fetchUserAccounts } from '../../actions/fetchUserAccountsAction';

export class ProfilePage extends Component {
  state = {
    user: {
      firstName: '',
      lastName: '',
      phoneNumber: '',
      email: '',
    },
  }

  componentDidMount = () => {
    const token = localStorage.getItem('token');
    if (!token) {
      this.props.history.push('/login', {});
      return;
    }
    const user = JSON.parse(localStorage.getItem('user'));
    this.setState({ user })
    const { email } = user;

    const { fetchAccounts } = this.props;
    fetchAccounts(email);
  };

  logout(){
    localStorage.removeItem('token');
}

renderAccount(account) {
  return (
    <div className="account-detail-container" key={account.accountnumber}>
    <table>
        <caption colSpan="2">{account.accountnumber} Details</caption>
     <tbody>
      <tr>
        <th>Acc. Number:</th>
        <td><i className="fa fa-none"></i>{account.accountnumber}</td>
      </tr>
      <tr>
        <th> Type:</th>
        <td><i className="fa fa-none"></i>{account.type}</td>
      </tr>
      <tr>
        <th>Status:</th>
        <td><i className="fa fa-check"></i> {account.status} </td>
      </tr>
      <tr>
        <th>Date Created:</th>
        <td><i className="fa fa-calendar-o"></i> {account.createdon}</td>
      </tr>
      <tr>
        <th>Balance:</th>
        <td><i className="fa fa-money"></i> {account.balance}</td>
      </tr>
      </tbody>

    </table>
  </div>
  )
}

  render() {
    const { user: { firstName, lastName, phoneNumber, email } } = this.state
    const { userAccounts: { isLoading, accounts } } = this.props;
    return (
  <Dashboard header_title="User Dashboard">
    <div className="dasboard-body">
      <div className="profile-container">
        <div className="personal-detail-container">
          <div className="profile-image">
            <div className="profile-pic">
              <img src={require('../../assets/image//user.png')} alt="" />
            </div>
          </div>
          <div className="profile-detail">
            <table>
                <caption colSpan="2">Personal Details</caption>
              <tbody>
              <tr>
                <th> Name:</th>
                <td>
                  <i className="fa fa-user" /> {firstName} {lastName}
                </td>
              </tr>
              <tr>
                <th>Phone Number:</th>
                <td>
                  <i className="fa fa-phone" /> {phoneNumber}
                </td>
              </tr>
              <tr>
                <th>Address:</th>
                <td>
                  <i className="fa fa-map-marker" /> Not Available
                </td>
              </tr>
              <tr>
                <th>Date of Birth:</th>
                <td>
                  <i className="fa fa-calendar-o" /> Not Available
                </td>
              </tr>
              <tr>
                <th>Email:</th>
                <td>
                  <i className="fa fa-envelope" /> {email}
                </td>
              </tr>
              <tr>
                <th>Next of kin:</th>
                <td>
                  <i className="fa fa-child" /> Not Available
                </td>
              </tr>
              </tbody>
            </table>
          </div>
        </div>
        
        {accounts.map(account => this.renderAccount(account))}
      </div>
     
    </div>
  </Dashboard>

);
}
}

const mapStateToProps = state => ({
  errors: state.accountReducer.errors,
  userAccounts: state.fetchAccountsReducer,
});

export const mapDispatchToProps = dispatch => ({
  fetchAccounts: (email) => dispatch(fetchUserAccounts(email)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ProfilePage);