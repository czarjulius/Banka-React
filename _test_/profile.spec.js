import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { ProfilePage } from '../src/components/Profile/ProfilePage';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const renderSignInPage = args => {
  const defaultProps = {
    history: {
        push: jest.fn()
    },
    match: {},
    userAccounts: {
      isLoading: false,
      accounts: []
    },
    fetchAccouts: jest.fn(),
    errors: undefined,
    location: {}
  };
  const props = { ...defaultProps, ...args };
  return mount(
    <BrowserRouter>
      <ProfilePage {...props} />
    </BrowserRouter>
  );
};

describe('Profile in Components Tests', () => {
  it('renders Profile In template', () => {
    const wrapper = renderSignInPage();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('DashboardPage').length).toBe(1);
  });
});