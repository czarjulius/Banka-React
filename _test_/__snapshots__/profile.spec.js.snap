// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile in Components Tests renders Profile In template 1`] = `ReactWrapper {}`;
